### PR TITLE
UCP/AM/MRAIL: added limitations for AM_BW lanes

### PR DESCRIPTION
**Pull request by hoopoepg**
- due to limitations in registration of memory on sender side
  there should be limited number of AM BW lanes
- minor optimization in AM BW lane init

### CODE REVIEW

File: src/ucp/wireup/select.c [hunk 1]
```diff
@@ -879,18 +881,24 @@ static ucs_status_t ucp_wireup_add_am_bw_lanes(ucp_ep_h ep, const ucp_ep_params_
...
+    md_map        = 0;
 
     /* am_bw_lane[0] is am_lane, so don't re-select it here */
     for (lane_desc_idx = 0; lane_desc_idx < *num_lanes_p; ++lane_desc_idx) {
         if (lane_descs[lane_desc_idx].usage & UCP_WIREUP_LANE_USAGE_AM) {
             addr_index     = lane_descs[lane_desc_idx].addr_index;
             rsc_index      = lane_descs[lane_desc_idx].rsc_index;
             remote_md_map &= ~UCS_BIT(address_list[addr_index].md_index);
+            md_map        |= UCS_BIT(context->tl_rscs[rsc_index].md_index);
             tl_bitmap      = ucp_wireup_unset_tl_by_md(ep, tl_bitmap, rsc_index);
```
Comment (hoopoepg) [inline src/ucp/wireup/select.c]: @yosefe shouldn't we break here? we always have only one AM lane

Comment (yosefe) [reply src/ucp/wireup/select.c]: yes we can break here but it works either way

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5911/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5912/ for details (Mellanox internal link).

**Pull request merged**